src/main/java/co/elastic/support/diagnostics/commands/GenerateLogstashDiagnostics.java

```diff
@@ -0,0 +1,171 @@
+ *  2.0; you may not use this file except in compliance with the Elastic License
+ *  2.0.
+ */
+package co.elastic.support.diagnostics.commands;
+
+import co.elastic.support.Constants;
+import co.elastic.support.diagnostics.DiagnosticException;
+import co.elastic.support.diagnostics.chain.Command;
+import co.elastic.support.diagnostics.chain.DiagnosticContext;
+import co.elastic.support.rest.RestClient;
+import co.elastic.support.rest.RestResult;
+import co.elastic.support.util.SystemUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateExceptionHandler;
+import freemarker.template.TemplateModelException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GenerateLogstashDiagnostics implements Command {
+
+    private static final Logger logger = LogManager.getLogger(GenerateLogstashDiagnostics.class);
+
+    private static final String DIAGNOSTIC_TEMPLATES_PATH = "logstash-diagnostic-templates/";
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Override
+    public void execute(DiagnosticContext context) throws DiagnosticException {
+        try {
+            final RestClient client = context.resourceCache.getRestClient(Constants.restInputHost);
+            final JsonData jsonData = new JsonData(getNodeInfo(client), getNodeStats(client));
+            final Configuration configuration = createTemplateConfiguration();
+
+            for (String templateFileName : getAllTemplates()) {
+                final String outputFileName = templateFileName.substring(0, templateFileName.lastIndexOf(".ftlh"));
+                final Path outputFilePath = Paths.get(context.tempDir, "logstash_diagnostic_" + outputFileName);
+                try {
+                    final Template template = configuration.getTemplate(templateFileName);
+                    final StringWriter writer = new StringWriter();
+                    template.process(new Datasource(jsonData), writer);
+                    SystemUtils.writeToFile(writer.toString(), outputFilePath.toString());
+                } catch (IOException | TemplateException e) {
+                    logger.error(Constants.CONSOLE, String.format("Could not generate Logstash diagnostic: %s - skipping template.", templateFileName), e);
```

Comment:
Why use `String.format` instead of leveraging logging framework placeholders?

Reply:
It's using the `error(Marker marker, String message, Throwable throwable)` method overload. I can switch it to the parametrized method to leverage the placeholder replacements, but TBH, I find the "exception on the last parameter" not so clear if you don't know L4J's that much.

Reply:
This code doesn't need to be so performant, but in general is best to use log4j placeholder, because does a lazy evaluation (if debug level is not match the substitution of the string element, or the "toString", is not executed) while using the `String.format` does an eager evaluation.